category_encoders/hashing.py

```diff
@@ -238,18 +161,64 @@ def _transform_single_cpu(self, X, override_return_df=False):
         if not list(self.cols):
             return X
 
-        X = self.hashing_trick(X, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
-
-        if self.drop_invariant:
-            X = X.drop(columns=self.invariant_cols)
+        X = self.hashing_trick(
+            X, 
+            hashing_method=self.hash_method,
+            N=self.n_components, 
+            cols=self.cols, 
+        )
+        
+        return X
 
-        if self.return_df or override_return_df:
-            return X
-        else:
-            return X.to_numpy()
+    @staticmethod
+    def hash_chunk(shm_result, np_df, N, shm_offset):
+        for i, row in enumerate(np_df):
+            for val in row:
+                if val is not None:
+                    hasher = hashlib.md5()
+                    hasher.update(bytes(str(val), 'utf-8'))
+                    column_index = int.from_bytes(hasher.digest(), byteorder='big') % N
+                    row_index = (shm_offset + i)*N
+                    shm_index = row_index + column_index
+                    shm_result[shm_index] += 1
+
+    def hashing_trick_with_np_parallel(self, df, N):
+        np_df = df.to_numpy()
+        shm_result = multiprocessing.RawArray(HashingEncoder.default_int_np_array.dtype.char, len(df)*N)
+
+        n_process = []
+        chunk_size = int(len(np_df)/self.max_process)
```

Comment:
this ignores the `max_samples` parameter and might lead to the process crashing in case of too much data / too few CPUs

Reply:
That is actually another advantage of using a shared memory instead of queues. 

In the previous implementation, there were 3 mains things that could take up memory:
1. the input dataframe, stored in `self.X` - it is copied accross all processes wether we fork or spawn because the process target is `self.require_data`. 
2. the output dataframe, which is the concatenation of all `hashing_parts`. There is only one instance of it in the main process. It takes space of exactly the output dataframe when all processes are done computing their chunk of the data.
3. the `data_part` that's temporary to each subprocess, and that's created to be put on the queue after the part is calculated.

In the worst case scenario of the queue implementation, the total memory usage is: `input_size * number_of_processes + output_size + (ouput_size/max_sample)*number_of_processes`

In the new implementation, only two things, because we write directly to the output shared memory instead of having the data transiting through the queue:
1. the input dataframe, stored in `self.X` - it is copied accross all processes wether we fork or spawn because the process has in its args `np_df`.
2. the output dataframe, which is the shared memory and is allocated only accross all processes.

The total memory usage in the shared memory implementation is: `imput_size * number_of_processes + output_size`

That means that the memory usage is always smaller with the shared memory implementation, and does not depend on `max_samples`